docs/source/basics.rst

```diff
@@ -273,14 +273,83 @@ be executed 1000 times.
 
 Parametric Compilation
 ~~~~~~~~~~~~~~~~~~~~~~
-*Coming soon*
+
+Keri to improve:
+[
+Making use of declared memory regions, we can load values to parametric gates at execution time, after compilation. We
+can specify how we want to patch our compiled executable binaries with updated parameters. For some quantum algorithms,
+it's necessary to run a program repeatedly, but with different parameter values each time.
+Taking the compiler out of the execution loop for programs like this offers a huge performance improvement over compiling
+the whole program each time with those values filled in.
+(Some more details about this and an example are found :ref:`here <parametric>`.)
+]
+
+The first step is to build our parametric program, which functions like a template for all the precise programs we will
+run. Below we create a simple example program to illustrate, which puts the qubit onto the equator of the Bloch Sphere and then
+rotates it around the Z axis for some variable angle theta before applying another X pulse and measuring.
+
+.. code:: python
+
+    import numpy as np
+
+    from pyquil import Program
+    from pyquil.gates import *
```

Comment:
no glob :(

Reply:
idk I've been using `from pyquil.gates import *` because it makes a little bit of sense here

 - that file has a well-specced `__all__`, so you don't get garbage
 - you might as well get all the gates

but maybe for docs it hides where things live